traffic_monitor/tests/integration/README.md

```diff
@@ -0,0 +1,12 @@
+# Traffic Monitor Integration Test Framework
+
+## Running
+
+From the `trafficcontrol/traffic_monitor` directory:
+
+```
+(cd tools/testto && go build)
+(cd tools/testcaches && go build)
+(cd tests/integration && go test -c -o traffic_monitor_integration_test)
+sudo docker-compose -p tmi --project-directory . -f tests/integration/docker-compose.yml run tmintegrationtest
```

Comment:
After resolving the above linking error, rebuilding the images, and re-running, I get unmarshalling errors like

```json
{"error": "unmarshalling posted body: json: cannot unmarshal bool into Go struct field CRConfigDeliveryService.deliveryServices.anonymousBlockingEnabled of type string"}
```

It eventually says

```
Error communicating with Monitor 'http://trafficmonitor' - didn't return a 200 OK in 30s
```

<details><summary>Full <code>tmintegrationtest</code> logs are here (click to expand)</summary>

```
< HTTP/1.1 200 OK
* About to connect() to testto port 80 (#0)
*   Trying 172.23.0.3...
* Connected to testto (172.23.0.3) port 80 (#0)
> POST /api/1.2/cdns/fake/snapshot HTTP/1.1
> User-Agent: curl/7.29.0
> Host: testto
> Accept: */*
> Content-Length: 3756
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
>
< HTTP/1.1 100 Continue
< HTTP/1.1 400 Bad Request
< Date: Fri, 25 Sep 2020 21:32:19 GMT
< Content-Length: 169
< Content-Type: text/plain; charset=utf-8
* HTTP error before end of send, stop sending
<
* Closing connection 0
{"error": "unmarshalling posted body: json: cannot unmarshal bool into Go struct field CRConfigDeliveryService.deliveryServices.anonymousBlockingEnabled of type string"}* About to connect() to testto port 80 (#0)
*   Trying 172.23.0.3...
* Connected to testto (172.23.0.3) port 80 (#0)
> POST /api/1.2/cdns/fake/configs/monitoring.json HTTP/1.1
> User-Agent: curl/7.29.0
> Host: testto
> Accept: */*
> Content-Length: 2308
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
>
< HTTP/1.1 100 Continue
< HTTP/1.1 400 Bad Request
< Date: Fri, 25 Sep 2020 21:32:19 GMT
< Content-Length: 167
< Content-Type: text/plain; charset=utf-8
* HTTP error before end of send, stop sending
<
* Closing connection 0
{"error": "unmarshalling posted body: json: cannot unmarshal object into Go struct field TrafficServer.trafficServers.deliveryServices of type []tc.tsdeliveryService"}* About to connect() to testto port 80 (#0)
*   Trying 172.23.0.3...
* Connected to testto (172.23.0.3) port 80 (#0)
> POST /api/1.2/servers HTTP/1.1
> User-Agent: curl/7.29.0
> Host: testto
> Accept: */*
> Content-Length: 1135
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
>
< HTTP/1.1 100 Continue
< HTTP/1.1 204 No Content
< Date: Fri, 25 Sep 2020 21:32:19 GMT
<
* Connection #0 to host testto left intact


testto:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3877    0  3877    0     0   657k      0 --:--:-- --:--:-- --:--:--  757k
{"response":{
  "config": {
    "api.cache-control.max_age": "30",
    "consistent.dns.routing": "true",
    "coveragezone.polling.interval": "30",


testcaches:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
{ 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
1  "ats": {
0    "plugin.remap_stats.num7.example.net.in_bytes": 2,
0    "plugin.remap_stats.num7.example.net.out_bytes": 2,
     "plugin.remap_stats.num7.example.net.status_2xx": 1,
34955    0 34955    0     0  4648k      0 --:--:-- --:--:-- --:--:-- 4876k
(23) Failed writing body


traffic_monitor:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6522<!DOCTYPE html>0     0      0      0 --:--:-- --:--:-- --:--:--     0

 <!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
0  6522    0     0  1159k      0 --:--:-- --:--:-- --:--:-- 1273k
(23) Failed writing body
DEBUG traffic_monitor_integration starting

Error communicating with Monitor 'http://trafficmonitor' - didn't return a 200 OK in 30s
```
</details>

Looking at the logs of the `trafficmonitor` container, it doen't seem to get past

```go
ERROR: datareq.go:152: 2020-09-25T21:32:38.143874758Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
```

although there are other errors in there too, like

```go
trafficmonitor_1     | ERROR: monitorconfig.go:346: 2020-09-25T21:32:33.530687712Z: Failed to parse polling strings for cache server 'server0': no service addresses found
trafficmonitor_1     | ERROR: monitorconfig.go:346: 2020-09-25T21:32:33.530798803Z: Failed to parse polling strings for cache server 'server1': no service addresses found
```

<details><summary>Full <code>trafficmonitor</code> container logs are here (click to expand)</summary>

```go
Attaching to tmi_trafficmonitor_1
trafficmonitor_1     | Failed to get D-Bus connection: Operation not permitted
trafficmonitor_1     | /etc/init.d/traffic_monitor: line 41: /etc/sysconfig/network: No such file or directory
trafficmonitor_1     | Starting traffic_monitor: 
trafficmonitor_1     | ERROR: opsconfig.go:77: 2020-09-25T21:32:18.530467624Z: OpsConfigManager: getting CDN name from Traffic Ops, using config CDN 'nocdn': getting monitor CDN: no server 'trafficmonitor' found in Traffic Ops
trafficmonitor_1     | 
trafficmonitor_1     | ERROR: opsconfig.go:77: 2020-09-25T21:32:18.538113399Z: OpsConfigManager: Error getting Traffic Ops data: Error getting CRconfig from Traffic Ops: invalid CRConfig: CRConfig.Stats.CDN missing
trafficmonitor_1     | 
trafficmonitor_1     | ERROR: opsconfig.go:205: 2020-09-25T21:32:18.538157142Z: retrying in 100ms
trafficmonitor_1     | ERROR: opsconfig.go:77: 2020-09-25T21:32:18.638750564Z: OpsConfigManager: Error getting Traffic Ops data: Error getting CRconfig from Traffic Ops: invalid CRConfig: CRConfig.Stats.CDN missing
trafficmonitor_1     | 
trafficmonitor_1     | ERROR: opsconfig.go:205: 2020-09-25T21:32:18.638781915Z: retrying in 248.612746ms
trafficmonitor_1     | ERROR: opsconfig.go:77: 2020-09-25T21:32:18.888066827Z: OpsConfigManager: Error getting Traffic Ops data: Error getting CRconfig from Traffic Ops: invalid CRConfig: CRConfig.Stats.CDN missing
trafficmonitor_1     | 
trafficmonitor_1     | ERROR: opsconfig.go:205: 2020-09-25T21:32:18.888119514Z: retrying in 642.263625ms
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:20.102734056Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:21.1049107Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:22.107247084Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:23.109295834Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:24.111672602Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:25.113719758Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:26.115813436Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:27.118294393Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:28.120237324Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:29.122398443Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:30.123841491Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:31.125970238Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:32.128329372Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:33.130950123Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: monitorconfig.go:346: 2020-09-25T21:32:33.530687712Z: Failed to parse polling strings for cache server 'server0': no service addresses found
trafficmonitor_1     | ERROR: monitorconfig.go:346: 2020-09-25T21:32:33.530798803Z: Failed to parse polling strings for cache server 'server1': no service addresses found
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:34.135402114Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:35.136748588Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:36.139031306Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:37.141564084Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:38.143874758Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:39.146134866Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:40.14928317Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:41.151503741Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:42.154148808Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:43.156421926Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:44.158672414Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:45.159938856Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:46.161990084Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:47.163412929Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:48.164121248Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
trafficmonitor_1     | ERROR: datareq.go:152: 2020-09-25T21:32:49.165983499Z: Request Error: /api/version: service still starting, some caches unpolled: map[]
```
</details>